HW2/P2/P2.txt

```diff
@@ -0,0 +1,5 @@
+Both parallel approaches were at least 30 times slower than serial code (0.3s vs 10-18s), presumably due to overhead. The fine-grained and medium-grained approaches took about the same amount of time.
+
+P2graph.png shows a plot of medium-grained locking performance for N from 1 to 20. For some reason, even values of N produced a better runtime than odd values. In addition, for correlated data, N=10 resulted in much worse performance than any other value of N, though this may be random noise.
```

Comment:
Great graph, makes it pretty clear what regions are ideal for each set of data. I wonder what a smoothed grapha averaged over multiple runs would look like